Configuration/Realm/Realm.xcconfig

```diff
@@ -33,7 +33,7 @@ REALM_PLATFORM_SUFFIX_macos = macosx;
 REALM_PLATFORM_SUFFIX_iosmac = maccatalyst;
 REALM_PLATFORM_SUFFIX = $(REALM_PLATFORM_SUFFIX_$(SDK_VARIANT));
 OTHER_LDFLAGS[sdk=macosx*] = -lrealm-$(REALM_PLATFORM_SUFFIX)$(REALM_LIBRARY_SUFFIX) $(REALM_CATALYST_FLAGS);
-OTHER_LIBTOOLFLAGS[sdk=macosx*] = -lrealm-$(REALM_PLATFORM_SUFFIX)$(REALM_LIBRARY_SUFFIX) $(REALM_CATALYST_FLAGS);
```

Comment:
Why do we no longer need the flags?

Reply:
Passing the `-target` flag to `ld` never actually worked and I just failed to test this variant of the build when I added that bit (it's only applicable to the xcframework build and not the dynamic frameworks that were already getting tested on CI).